### ISSUE TITLE
support address parameters in channels

### ISSUE THREAD

**Issue by KhudaDad414** [asyncapi/glee#544]
### Reason/Context
suppose I have the following AsyncAPI spec file in version 3 of the spec:

```yaml
...
servers:
  slack:
    host: wss-primary.slack.com
    protocol: wss
channels:
  main:
    address: "/link/?ticket={ticket}&app_id={appID}"
    parameters:
      ticket:
        description: "your ticket ID. Ticket id can only be used for once."
        location: "$message.payload#/messageId"
        enum: ["a", "b"]
        default: ""
      appID:
        description: "Your slack app ID"
        enum: ["ffff", "ddddd"]
        default: ""
    bindings: 
      ws: {}
    messages:
      reactionAdded:
        $ref: "#/components/messages/reactionAdded"
...
```
it will fail because the [parameters](https://www.asyncapi.com/docs/reference/specification/v3.0.0-next-major-spec.14#parametersObject) are going to be ignored.
### Description
Glee should be able to handle the parameters.
#### If there is a `location`

parse the values from the message's payload or header and pass it to the operation function.
the new glee function type should be:
|Attribute|Description|
|----|----|
|payload|The payload/body of the received event.
|parameters|The parameters of the channel's address.
|headers|The headers/metadata of the received event.
|channel|The name of the channel/topic from which the event was read.
|serverName|The name of the server/broker from which the event was received.

> NOTE: glee should pick up the default value and use it if `location` is missing.

> NOTE: if `enum` is specified, the parameter should be validated.

**Comment by p21nc3** [comment]
Hi @KhudaDad414, this can be fixed by making changes to parse parameters, having a new function type with attributes (payload, parameters, header, channels, servername). additionally enum validation & default value handling. here introducing an function type may or may not have regression in the framework, will run some test to validate that.
I would like to work on it, requesting to assign this issue to me.

**Comment by KhudaDad414** [comment]
@p21nc3 we do not assign issues. you can start working on it. if you need some help please let me know.

**Comment by KhudaDad414** [comment]
@p21nc3 I have started working on the issue here: #551 please check it out and leave some feedback.

**Comment by KhudaDad414** [comment]
/progress 60
partially implemented in #551
